actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -750,10 +769,45 @@ func findByChannelLookup(ctx context.Context, storage exec.Storage, byPayer exec
 
 // checkCondition combines params in the condition with the redeemerSuppliedParams, sends a message
 // to the actor and method specified in the condition, and returns an error if one exists.
-func checkCondition(vmctx exec.VMContext, condition *types.Predicate, redeemerSuppliedParams []interface{}) (uint8, error) {
+func checkCondition(vmctx exec.VMContext, chid *types.ChannelID, payerAddress address.Address, condition *types.Predicate, redeemerSuppliedParams []interface{}) (uint8, error) {
 	if condition == nil {
 		return 0, nil
 	}
+
+	if len(redeemerSuppliedParams) == 0 {
+		ctx := context.Background()
+		storage := vmctx.Storage()
+
+		var cachedCondition *types.Predicate
+		err := withPayerChannelsForReading(ctx, storage, payerAddress, func(byChannelID exec.Lookup) error {
```

Comment:
Instead of fetching the payment channel twice, move the condition check into updateCondition with the other validations. We can set the cached condition there as well.